### PR TITLE
Added docs on Indexing events

### PR DESCRIPTION
**Pull request by nikcio**
Adds to #191 

Also added a XML comment to the `LuceneIndex.IndexCommitted` event

### CODE REVIEW

File: docs/indexing.md [hunk 1]
```diff
@@ -189,6 +191,115 @@ If using Examine with the default Lucene implementation then the `IIndex` implem
...
+
+index.TransformingIndexValues += TransformingIndexValues;
+
+private void TransformingIndexValues(object sender, IndexingItemEventArgs e)
+{
+    // Customize the ValueSet
+}
+```
+
+### BaseIndexProvider.IndexingError
```
Comment (Shazwazza) [inline docs/indexing.md:237]: This is actually part of IIndex in v3

File: docs/indexing.md [hunk 2]
```diff
@@ -189,6 +191,115 @@ If using Examine with the default Lucene implementation then the `IIndex` implem
...
+
+index.IndexOperationComplete += IndexOperationComplete;
+
+private void IndexOperationComplete(object sender, IndexOperationEventArgs e)
+{
+    // Index operation completed
+}
+```
+
+### BaseIndexProvider.TransformingIndexValues
```
Comment (Shazwazza) [inline docs/indexing.md:217]: This is actually part of IIndex in v3

**Review by Shazwazza — CHANGES_REQUESTED** [review]
Thanks! Have left a couple comments

**Comment by nikcio** [comment]
@Shazwazza Should be fixed now 😄

**Comment by nikcio** [comment]
@Shazwazza I've updated the docs changes to the new docfx from #322 so it's ready to merge now. It's been a while since this PR was created so  I'll just refresh what's going on:

- This PR adds code examples of how you can listen to indexing events.

**Comment by nikcio** [comment]
Hey @Shazwazza can i do anything to move this PR forward? 😁

**Comment by Shazwazza** [comment]
@nikcio  Thanks so much for all your effort :)

**Comment by Shazwazza** [comment]
I'll see if i can cherry pick this to the feature/docfx branch since that is where the docs are published from

File: docs/indexing.md [hunk 3]
```diff
@@ -142,7 +142,9 @@ Data is easily deleted from the index by the unique identifier you provided in y
 
 <!-- Tabs -->
 <div class="container">
-  <input type="radio" id="tab-link-20" name="docwriting" checked />
```
Comment (Shazwazza) [inline docs/indexing.md:145]: This doc doesn't merge with the feature/docfx branch where the docs now live so this wasn't cherry picked. For doc updates now, They'll have to be done against feature/docfx for now.

Comment (nikcio) [reply docs/indexing.md:145]: This should be fine the other file will be the one that will come on the docfx site😉

**Pull request merged**